### PR TITLE
Fixed bazel configuration ambiguity for non HTTP3 builds

### LINKED ISSUE #34482: Failure building for Linux PPC or Windows x64 with HTTP3 disabled
The `source/exe/BUILD` & `test/config_test/BUILD` files use `select()` to control the list of extensions to be included/excluded, and under certain circumstances, they can fail.

```
    deps = [
        .......
    ] + select({
        "//bazel:windows_x86_64": envoy_all_extensions(WINDOWS_SKIP_TARGETS),
        "//bazel:linux_ppc": envoy_all_extensions(PPC_SKIP_TARGETS),
        "//bazel:disable_http3": envoy_all_extensions(NO_HTTP3_SKIP_TARGETS),
        "//conditions:default": envoy_all_extensions(),
    }),
```

The problem is the list of conditions in the `select()` covers the two _orthogonal_ concerns of platform & HTTP3. This can lead to ambiguity and therefore failure if more than one option is true. In particular, if `//bazel:disable_http3 == True` then any attempt to build for `//bazel:windows_x86_64` or `//bazel:linux_ppc` will fail with the following message from bazel:

```
ERROR: /home/envoy/check_openssl/envoy-openssl/source/exe/BUILD:35:17: Illegal ambiguous match on configurable attribute "deps" in //source/exe:envoy_common_lib:
//bazel:linux_ppc
//bazel:disable_http3
Multiple matches are not allowed unless one is unambiguously more specialized or they resolve to the same value. See https://bazel.build/reference/be/functions#select.
INFO: Repository build_bazel_rules_swift instantiated at:
  /home/envoy/check_openssl/envoy-openssl/WORKSPACE:30:25: in <toplevel>
  /home/envoy/check_openssl/envoy-openssl/bazel/dependency_imports.bzl:32:29: in envoy_dependency_imports
  /root/.cache/bazel/_bazel_root/0cab9e149d5530907041b1f9089d77ad/external/build_bazel_rules_apple/apple/repositories.bzl:124:15: in apple_rules_dependencies
  /root/.cache/bazel/_bazel_root/0cab9e149d5530907041b1f9089d77ad/external/build_bazel_rules_apple/apple/repositories.bzl:86:14: in _maybe
Repository rule http_archive defined at:
  /root/.cache/bazel/_bazel_root/0cab9e149d5530907041b1f9089d77ad/external/bazel_tools/tools/build_defs/repo/http.bzl:372:31: in <toplevel>
ERROR: Analysis of target '//:envoy' failed; build aborted:
```

### PR DESCRIPTION
**Pull request by tedjpoole**
To avoid ambiguity when building with `//bazel:http3=False` (on PPC and Windows) this commit adds 2 new `config_setting_group`s and updates the relevant `select()` statements.

Fixes #34482

### CODE REVIEW

**Review by phlax — APPROVED** [review]
lgtm, thanks @tedjpoole

**Review by phlax — DISMISSED** [review]

**Comment by tedjpoole** [comment]
Sorry but this fix isn't complete. I have another commit coming in a few minutes...

File: bazel/BUILD [hunk 1]
```diff
@@ -351,6 +351,64 @@ selects.config_setting_group(
...
+    name = "always_false_flag",
+    build_setting_default = False,
+    visibility = ["//visibility:private"],
+)
+
+# Alias equal to "not(":disable_http3")" (if "not()" existed).
+alias(
+    name = "enable_http3_setting",
+    actual = select({
+        ":disable_http3": ":always_false_flag",
```
[the remark this answers is not in the document]
Comment (tedjpoole) [reply bazel/BUILD:386]: It fails if the literal values are used directly. It is expecting a string (a rule name to be aliased).
```
ERROR: /home/envoy/envoy-openssl/bazel/BUILD:384:6: //bazel:enable_http3_setting: expected value of type 'string' for each branch in select expression of attribute 'actual' in 'alias' rule (including '//bazel:disable_http3'), but got False (bool)
ERROR: /home/envoy/envoy-openssl/bazel/BUILD:384:6: //bazel:enable_http3_setting: missing value for mandatory attribute 'actual' in 'alias' rule
ERROR: Skipping '//bazel:http3': Error evaluating '//bazel:http3': error loading package 'bazel': Package 'bazel' contains errors
WARNING: Target pattern parsing failed.
ERROR: //bazel:http3 :: Error loading option //bazel:http3: Error evaluating '//bazel:http3': error loading package 'bazel': Package 'bazel' contains errors
```

[the remark this answers is not in the document]
Comment (phlax) [reply bazel/BUILD:386]: ah k - i see - its what the alias is expecting - perhaps we could rename these to something like something like `http3_enabled`/`http3_disabled` - or even just `enabled` etc - i think i was partly confused by the name

**Review by phlax — APPROVED** [review]
lgtm, thanks @tedjpoole

**Comment by tedjpoole** [comment]
@phlax Do you know if the test failures on this are known? They don't look related to the changes

**Comment by phlax** [comment]
yeah - it was fixed already i think - ill push this through

**Pull request merged**